packages/rxjs/src/internal/observable/dom/WebSocketSubject.ts

```diff
@@ -151,47 +148,53 @@ const WEBSOCKETSUBJECT_INVALID_ERROR_OBJECT =
 
 export type WebSocketMessage = string | ArrayBuffer | Blob | ArrayBufferView;
 
-export class WebSocketSubject<T> extends AnonymousSubject<T> {
-  private _config!: WebSocketSubjectConfig<T>;
-
-  /** @internal */
-  _output!: Subject<T>;
+export class WebSocketSubject<In, Out = In> extends Observable<Out> {
+  private _config: WebSocketSubjectConfig<In, Out> &
+    Required<Pick<WebSocketSubjectConfig<In, Out>, 'WebSocketCtor' | 'serializer' | 'deserializer'>>;
 
   private _socket: WebSocket | null = null;
 
-  constructor(urlConfigOrSource: string | WebSocketSubjectConfig<T> | Observable<T>, destination?: Observer<T>) {
+  private _inputBuffer: In[] = [];
+
+  private _hasError = false;
+
+  private _error: any;
+
+  private _isComplete = false;
+
+  private _subscriberCounter = 0;
+
+  private _subscribers = new Map<number, Subscriber<Out>>();
```

Comment:
Should these be real private fields without the ugly `_` prefix instead of TS private?

Reply:
Yeah. I think overall, RxJS is due for an overhaul in this area. The upside is we can't accidentally publicly expose any `#private` properties. The downside is you can't destructure them. Which is mostly a superficial downside.

Reply:
@benlesh, I might try to do this.